### PR TITLE
Add CPU m4.10xlarge instance type for Inductor CPU FP32 accuracy test with AVX2

### PR DESCRIPTION
**Pull request by zxd1997066**
Due to issue [#123224](https://github.com/pytorch/pytorch/issues/123224), we plan to add m4.10xlarge instance for testing inductor CPU FP32 accuracy with AVX2 ISA in Pytorch CI. We collected the time cost for running FP32 accuracy test on both m4.10xlarge and c5.12xlarge, and here is the results.
</head>
<body link=blue vlink=purple>

Config | Runner | Instance type | ISA | Price $/h | TB Acc /h | HF Acc /h | TIMM Acc /h | Total Time /h
-- | -- | -- | -- | -- | -- | -- | -- | -- 
1 | add in this PR | m4.10xlarge | AVX2 | 2 | 1.5 | 0.8 | 1.2 | 3.5
2 | linux.12xlarge | c5.12xlarge | AVX512, AVX2 | 2.04 | 1.5 | 0.5 | 0.6 | 2.6

</body>
</html>

Config 2 depends on PR [#123514](https://github.com/pytorch/pytorch/pull/123514) land, and if the inductor generates code which call oneDNN kernels, those kernels may not run with AVX2. So for AVX2 inductor tests, we prefer to add AVX2 only instance as new runner type. 

As we mentioned in PR #5089,  the average launch times/day of inductor cpu test is 174, and we can assume 70% ~ 100% tests can be completed. The base pricing of m4.10xlarge is **2 USD per hour**, so the average base cost of accuracy test running on m4.10xlarge is **852.6 ~ 1218**(174 * 3.5 * 2) USD per day

### CODE REVIEW

File: .github/scale-config.yml
Comment (chuanqi129) [inline .github/scale-config.yml:30]:
```suggestion
  linux.10xlarge.avx2:
```

**Comment by chuanqi129** [comment]
Hi @tylertitsworth, would you please help to add corresponding configs for ARC part?

**Review by atalman — COMMENTED** [review]
lgtm, do you really need max_available:  60 instances ? Should we keep more conservative 30 instances as with linux.24xl.spr-metal ?

**Comment by chuanqi129** [comment]
> lgtm, do you really need max_available: 60 instances ? Should we keep more conservative 30 instances as with linux.24xl.spr-metal ?

Thanks @atalman for the review and comments. Because the inductor test on this runner type need spend more time than SPR, about 2 times, refer PR #5089. To avoid potential long CI test queue of this test, we enlarged the max_avaliable number.

**Comment by tylertitsworth** [comment]
> Hi @tylertitsworth, would you please help to add corresponding configs for ARC part?

I need write access to your fork to contribute. If you have a better idea of how I should contribute I'm open to suggestions.

**Comment by zxd1997066** [comment]
> > Hi @tylertitsworth, would you please help to add corresponding configs for ARC part?
> 
> I need write access to your fork to contribute. If you have a better idea of how I should contribute I'm open to suggestions.

Hi @tylertitsworth , I added the access for you.
image

**Review by atalman — APPROVED** [review]

**Comment by tylertitsworth** [comment]
@zxd1997066 @chuanqi129 @atalman I've updated the config for Arc, please take a look.

**Review by chuanqi129 — APPROVED** [review]

**Review by ZainRizvi — APPROVED** [review]

**Comment by chuanqi129** [comment]
@atalman @ZainRizvi  Thanks for the review and comments, who can help on the Vercel check and merge this PR if there is no more comments? After that we can add AVX2 inductor test based on those AVX2 runners. cc: @malfet

**Comment by jeanschmidt** [comment]
Hi @zxd1997066

we're having alerts firing in regards to the max number of instances being hit for this particular runner type.

It seems that 60 is too low for the planned use case. Are you able to provide a analysis of the expected maximum number of runners that you can possibly hit and submit a PR?

In the meantime, I'll be suppressing alerts for 24h, please let me know if you need more time to sort this out.

**Comment by zxd1997066** [comment]
> Hi @zxd1997066
> 
> we're having alerts firing in regards to the max number of instances being hit for this particular runner type.
> 
> It seems that 60 is too low for the planned use case. Are you able to provide a analysis of the expected maximum number of runners that you can possibly hit and submit a PR?
> 
> In the meantime, I'll be suppressing alerts for 24h, please let me know if you need more time to sort this out.

Hi @jeanschmidt , thanks for the notice, I provided the analysis and submitted a PR for it https://github.com/pytorch/test-infra/pull/5514

**Pull request merged**
